libass/ass_rasterizer.c

```diff
@@ -683,6 +714,33 @@ static inline void rasterizer_fill_halfplane(const BitmapEngine *engine,
     }
 }
 
+enum {
+    FLAG_SOLID     = 1,
+    FLAG_COMPLEX   = 2,
+    FLAG_REVERSE   = 4,
+    FLAG_GENERIC   = 8,
+};
+
+static inline int get_fill_flags(struct segment *line, size_t n_lines, int winding)
+{
+    if (!n_lines)
+        return winding ? FLAG_SOLID : 0;
+    if (n_lines > 1)
+        return FLAG_COMPLEX | FLAG_GENERIC;
+
+    static const int test = SEGFLAG_UL_DR | SEGFLAG_EXACT_LEFT;
+    if (((line->flags & test) != test) == !(line->flags & SEGFLAG_DN))
+        winding++;
+
+    int flags = 0;
+    if (winding) flags ^= 1;
+    if (winding - 1) flags ^= 3;
+    static const int result[] = {
+        0, FLAG_COMPLEX, FLAG_SOLID, FLAG_COMPLEX | FLAG_REVERSE
+    };
+    return result[flags];
```

Comment:
Using nested ifs? I don't think that will be more clearer anyway but I consider it.